### PR TITLE
fix: manual receive address on Ledger when no buy accounts are added

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fixes a bug where the manual address entry input would not be shown if no accounts are connected for said buy asset on Ledger.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

> What protocols, transaction types or contract interactions might be affected by this PR?

Very low, the only risk here is the manual receive address entry may not be shown on other wallets - tested with MM and confirmed no regression

## Testing

- Connect EVM accounts only
- Try and trade to e.g BTC or RUNE
- Ensure you can input a manual receive address

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

- Develop

- This diff

### CODE REVIEW

**Comment by gomesalexandre** [comment]
* **#6420** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6420?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 1]
```diff
@@ -531,7 +542,7 @@ export const TradeInput = memo(({ isCompact }: TradeInputProps) => {
         >
           <WithLazyMount shouldUse={Boolean(receiveAddress)} component={RecipientAddress} />
           <WithLazyMount
-            shouldUse={hasUserEnteredAmount && !walletSupportsBuyAssetChain}
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:534]: This removal isn't a mistake:

1. feels a bit weird for this not to be shown initially from a UX perspective
2. this is wrongly reactive when there are no buy accounts and the trade buy asset doesn't display, and will cause issues

**Review by kaladinlight — APPROVED** [review]
Working as expected, only thought is that we have dupe logic in both `TradeInput` and `ManualAddressEntry` which feels weird. Seems like logic could be removed from `ManualAddressEntry` and showing the component would be controlled by `TradeInput`.

**Comment by gomesalexandre** [comment]
> Working as expected, only thought is that we have dupe logic in both `TradeInput` and `ManualAddressEntry` which feels weird. Seems like logic could be removed from `ManualAddressEntry` and showing the component would be controlled by `TradeInput`.

As discussed, will tackle that as a follow-up as this is not a change from the current ways we do things, but agree!

**Pull request merged**